### PR TITLE
Add bt cancel discovery.

### PR DESCRIPTION
**Pull request by xpconanfan**
This change is [Reviewable](https://reviewable.io/reviews/google/mobly-bundled-snippets/98)

### CODE REVIEW

**Comment by xpconanfan** [comment]
@jon-mann

File: src/main/java/com/google/android/mobly/snippet/bundled/bluetooth/BluetoothAdapterSnippet.java [hunk 1]
```diff
@@ -214,6 +214,30 @@ public void btBecomeDiscoverable(Integer duration) throws Throwable {
...
+        BroadcastReceiver receiver = new BluetoothScanReceiver();
+        mContext.registerReceiver(receiver, filter);
+        try {
+            if (!mBluetoothAdapter.cancelDiscovery()) {
+                throw new BluetoothAdapterSnippetException(
+                        "Failed to initiate to cancel bluetooth discovery.");
+            }
+            if (!Utils.waitUntil(() -> mIsScanResultAvailable, 120)) {
+                throw new BluetoothAdapterSnippetException(
+                        "Failed to get discovery results after 2 mins, timeout!");
```
Comment (jon-mann) [inline src/main/java/com/google/android/mobly/snippet/bundled/bluetooth/BluetoothAdapterSnippet.java]: if we're cancelling the discovery session, why do we care about the scan results?

File: src/main/java/com/google/android/mobly/snippet/bundled/bluetooth/BluetoothAdapterSnippet.java [hunk 2]
```diff
@@ -214,6 +214,30 @@ public void btBecomeDiscoverable(Integer duration) throws Throwable {
...
+        IntentFilter filter = new IntentFilter(BluetoothAdapter.ACTION_DISCOVERY_FINISHED);
+        mIsScanResultAvailable = false;
+        BroadcastReceiver receiver = new BluetoothScanReceiver();
+        mContext.registerReceiver(receiver, filter);
+        try {
+            if (!mBluetoothAdapter.cancelDiscovery()) {
+                throw new BluetoothAdapterSnippetException(
+                        "Failed to initiate to cancel bluetooth discovery.");
+            }
+            if (!Utils.waitUntil(() -> mIsScanResultAvailable, 120)) {
```
Comment (jon-mann) [inline src/main/java/com/google/android/mobly/snippet/bundled/bluetooth/BluetoothAdapterSnippet.java]: if we're cancelling the discovery session, why do we care about the scan results?

Comment (xpconanfan) [reply src/main/java/com/google/android/mobly/snippet/bundled/bluetooth/BluetoothAdapterSnippet.java]: Ah, sorry about the confusion, this var is really used to signify the completion of discovery.
Changed name.

**Pull request merged**